syntax = "proto3";

package nnstreamer.protobuf;

message Tensor {
  string name = 1;
  enum Tensor_type {
    NNS_NONE = 0;
    NNS_INT32 = 1;
    NNS_UINT32 = 2;
    NNS_INT16 = 3;
    NNS_UINT16 = 4;
    NNS_INT8 = 5;
    NNS_UINT8 = 6;
    NNS_FLOAT64 = 7;
    NNS_FLOAT32 = 8;
    NNS_INT64 = 9;
    NNS_UINT64 = 10;
  }
  Tensor_type type = 2;
  repeated uint32 dimension = 3;
  bytes data = 4;
}

message Tensors {
  uint32 num_tensor = 1;
  // frame_rate := rate_n / rate_d and it may be 0
  message frame_rate {
    int32 rate_n = 1;
    int32 rate_d = 2;
  }
  frame_rate fr = 2;
  repeated Tensor tensor = 3;
}
